modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -168,102 +183,124 @@ protected IndexRequest buildIndexRequest(SearchHit doc) {
          * Override the simple copy behavior to allow more fine grained control.
          */
         @Override
-        protected void copyRouting(IndexRequest index, SearchHit doc) {
+        protected void copyRouting(ActionRequest<?> request, String routing) {
             String routingSpec = mainRequest.getDestination().routing();
             if (routingSpec == null) {
-                super.copyRouting(index, doc);
+                super.copyRouting(request, routing);
                 return;
             }
             if (routingSpec.startsWith("=")) {
-                index.routing(mainRequest.getDestination().routing().substring(1));
+                super.copyRouting(request, mainRequest.getDestination().routing().substring(1));
                 return;
             }
             switch (routingSpec) {
-            case "keep":
-                super.copyRouting(index, doc);
-                break;
-            case "discard":
-                index.routing(null);
-                break;
-            default:
-                throw new IllegalArgumentException("Unsupported routing command");
+                case "keep":
```

Comment:
Nope! There are heathens who like it with indented case statements and they are in the majority too! But we don't have rules for it.